### ISSUE TITLE
Question: kPIDs 

### ISSUE THREAD

**Issue by cobysy** [osakechan/notoriousPID#6]
How did you arrive at the PID constants for the cooling and heating. 
Do I need to tune them myself before putting into use?

**Comment by osakechan** [comment]
Yes, those constants are unique to each installation and must be determined
experimentally (this is the largest challenge in configuring a PID
algorithm).  I highly recommend reading through Practical Process Control
by Doug J. Cooper (aka controlguru).  It can be found online at
http://www.controlguru.com/pages/table.html.  It is an excellent resource
and details methods for determination of the constants through a procedure
known as a bump test.  Allow the system to reach a steady state condition
at constant PID output, then 'bump' the process by changing the PID output
and observe the response of the system.  The data logging capabilities make
it easy to record this response.  It is relatively straight forward to
determine the constants from the response curve with a few simple
equations.  Good luck and Cheers!

--Mike aka Osakechan

**Comment by cobysy** [comment]
Thank you for finding the time to write the explanation osakechan (is that from お酒ちゃん I wonder?). Especially the part about the two PID algorithms I really appreciate. Because I could never find a good resource about how to PID a fridge compressor because, as you say, it's either fully ON or OFF. In fact the same goes for the resistive heat element (aquarium heater on a glass pitcher filled with water).

I'm still waiting for a few parts to arrive - hopefully I can get it all set up soon (and before my new work). Otherwise the STC1000 will have to do.

**Comment by divya0056** [comment]
Hi, I am a newbie learning more about this awesome project of yours. Wondering if you have a basic version of your project without data logging feature. So I can flash it on UNO rev3. 
I am trying to make a yougurt maker and any help from your end would be much appreciated.
Thanks 
Divya0056@gmail.com
